Makefile

```diff
@@ -269,12 +273,31 @@ coverage: clean-coverage
 
 test: test-cpp test-java
 
-build/$(TEST_JAR): $(TEST_SOURCES) build/$(CONVERTER_JAR)
+build/$(TEST_JAR): build/$(API_JAR) $(TEST_SOURCES) build/$(CONVERTER_JAR) $(TEST_DEPS_DIR) $(TEST_STUBS_DIR)
+	rm -rf build/test/classes
 	mkdir -p build/test/classes
-	$(JAVAC) -source $(JAVA_TARGET) -target $(JAVA_TARGET) -Xlint:-options -XDignore.symbol.file \
-	         -cp "build/jar/*:build/converter/*" -d build/test/classes $(TEST_SOURCES)
+	$(JAVAC) -source $(JAVA_TARGET) -target $(JAVA_TARGET) -Xlint:-options -cp "build/jar/*:$(TEST_DEPS_DIR)/*:$(TEST_GEN_DIR)/*:$(TEST_STUBS_DIR)/*" -d build/test/classes $(TEST_SOURCES)
```

Comment:
Please resolve merge conflict, you've overwritten my previous change.

Reply:
412ed552bc1798d0796c9cbfad716d6439da952a